app/src/main/java/org/simple/clinic/drugs/EditMedicinesEffectHandler.kt

```diff
@@ -42,9 +42,13 @@ class EditMedicinesEffectHandler @AssistedInject constructor(
 
       val protocolDrugsStream = effects
           .observeOn(io)
-          .flatMap { userSession.requireLoggedInUser() }
-          .switchMap { facilityRepository.currentFacility(it) }
+          .map {
+            val loggedInUser = userSession.loggedInUserImmediate()!!
+            facilityRepository.currentFacilityImmediate(loggedInUser)!!
```

Comment:
What we actually need here is the protocol UUID, so what you think about extracting this to a separate method that returns the protocol UUID instead?

Reply:
We do not need `facility` anywhere else. So I think its alright to keep the `protocolUuid` extraction here itself.

Reply:
Yeah, that's what I meant too. If we have a separate method in the effect handler

```kotlin
fun getCurrentProtocolUuid(): UUID {
  // get protocol UUID here
}
```

Then the implementation detail that the protocol ID is being fetched via the current facility is hidden from the Rx chain.

But this is just a suggestion. If you feel that the current implementation is fine, that's ok too.